### DISCUSSION TITLE
[Feature] Request a feature to compress selected image

### DISCUSSION THREAD

**Discussion by rddewan** [fluttercandies/flutter_wechat_assets_picker#322] [Ideas]
It would be good if the plugin can support file compression , currently we need to compress the file after picking the image

**Comment by AlexV525** [comment]
The main aspect of the package is to provide a flexible picking experience for most use cases. Thus operations related to I/O (conversion, compression, etc) take no place in the milestone. Also, the feature can be implemented in two paths from simple to complex:

- Use `AssetEntity.thumbnailDataWithSize` for precise control when generating thumbnails, which can adjust image size, format, and quality. When specific file sizes are required, it can be called recursively or using `while` to obtain thumbnails until it matches your requirement.
- Implement your own `AssetPickerBuilderDelegate` and only overrides widgets that you want to customize with a compress button or other widgets. This is more convenient after Dart 2.17 which supports **Super arguments** that can significantly reduce the template code when extending the builder delegate.

**Reply by rddewan** [reply]
thank you let me try

**Reply by bonaparta13** [reply]
you can use flutter_native_image  package to compress selected image
